api/src/services/schema.test.ts

```diff
@@ -0,0 +1,331 @@
+import knex, { Knex } from 'knex';
+import { getTracker, MockClient, Tracker } from 'knex-mock-client';
+import { afterEach, beforeAll, describe, expect, it, vi } from 'vitest';
+import { SchemaService } from '.';
+import { ForbiddenException } from '..';
+import { Snapshot, SnapshotDiffWithHash } from '../types/snapshot';
+import { applyDiff } from '../utils/apply-diff';
+import { getSnapshot } from '../utils/get-snapshot';
+
+vi.mock('../../package.json', () => ({ version: '0.0.0' }));
+
+vi.mock('../../src/database/index', () => {
+	return { __esModule: true, default: vi.fn(), getDatabaseClient: vi.fn().mockReturnValue('postgres') };
+});
+
+vi.mock('../utils/get-snapshot', () => ({
+	getSnapshot: vi.fn(),
+}));
+
+vi.mock('../utils/apply-diff', () => ({
+	applyDiff: vi.fn(),
+}));
+
+class Client_PG extends MockClient {}
+
+let db: Knex;
+let tracker: Tracker;
+
+beforeAll(() => {
+	db = knex({ client: Client_PG });
+	tracker = getTracker();
+});
+
+afterEach(() => {
+	tracker.reset();
+	vi.clearAllMocks();
+});
+
+describe('Services / Schema', () => {
+	describe('snapshot', () => {
+		it('should throw ForbiddenException for non-admin user', async () => {
+			const testSnapshot = {
```

Comment:
Can we dedup all those test data (in this case with line 59)? Same with `snapshotDiffWithHash`, ...

Reply:
I did thought about it, but there was some discussion on making tests more "contained" within themselves (even if it results in dupes) 🤔 That being said, it's probably less important here so I'll proceed to dedupe them 👍

Reply:
Okay, I see! Whatever feels right for you 👍😃

Reply:
Done! 🙌

Reply:
Nice 🙌  Thanks!